sql/hive/src/test/scala/org/apache/spark/sql/hive/HiveSparkSubmitSuite.scala

```diff
@@ -87,13 +88,16 @@ class HiveSparkSubmitSuite
     runSparkSubmit(args)
   }
 
-  ignore("SPARK-8489: MissingRequirementError during reflection") {
+  test("SPARK-8489: MissingRequirementError during reflection") {
     // This test uses a pre-built jar to test SPARK-8489. In a nutshell, this test creates
     // a HiveContext and uses it to create a data frame from an RDD using reflection.
     // Before the fix in SPARK-8470, this results in a MissingRequirementError because
     // the HiveContext code mistakenly overrides the class loader that contains user classes.
     // For more detail, see sql/hive/src/test/resources/regression-test-SPARK-8489/*scala.
-    val testJar = "sql/hive/src/test/resources/regression-test-SPARK-8489/test.jar"
+    import Properties.versionString
+    val version = versionString.substring(versionString.indexOf(" ") + 1,
```

Comment:
it might be better if you explicitly match 2.10 and 2.11, and throw exceptions for other things. I think the error message then would be more obvious when we introduce support for 2.12.